src/hooks/useRecommendationsObserver.ts

```diff
@@ -0,0 +1,68 @@
+import { useEffect } from 'react';
+import { Nullable } from '@constructor-io/constructorio-client-javascript/lib/types';
+import ConstructorIO from '@constructor-io/constructorio-client-javascript';
+import { Section } from '../types';
+
+/**
+ * Custom hook that observes the visibility of recommendation sections and calls trackRecommendationView event.
+ * This is done by using the IntersectionObserver API to observe the visibility of each recommendation section.
+ * That is done by passing the ref of each recommendation section to the IntersectionObserver.
+ * The refs are either passed by the user in section config or generated by the library by default.
```

Comment:
vocab alignment:
 - "user" - end user who's browser this production code is rendered on
 - "lib consumer" - developer / team who is consuming this library and building on top of it